test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/BasicSearchTests.cs

```diff
@@ -199,6 +199,34 @@ public async Task GivenTooBigPostRequest_WhenSearching_ThenDontCrashServer()
             await Client.SearchPostAsync("Patient", null, default, ("name", sb.ToString()));
         }
 
+        /// <summary>
+        /// Test to make sure we return Bad Request when customer sends too many parameters
+        /// </summary>
+        /// <returns>Task</returns>
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        [HttpIntegrationFixtureArgumentSets(dataStores: DataStore.SqlServer)]
+        public async Task GivenTooParametersPostRequest_WhenSearching_ThenBadRequestIsReturned()
```

Comment:
nit: naming could be a little clearer 
```suggestion
        public async Task GivenTooManyParametersInAPostRequest_WhenSearching_ThenBadRequestIsReturned()
```